### DISCUSSION TITLE
The useTable with customFilter with not automatically fetch data to server

### DISCUSSION THREAD

**Discussion by heriparid** [refinedev/refine#3858] [Q&A]
Hi Refine Dev,
I encountered an issue when trying to implement useTable with custom filter.
My use case is that:
I want to build a table with custom filter, and then when the first page loaded, the table will not automatically running to fetch the data.
I read the documentation by simple set the queryOptions.enabled to false.
This my snippet code:
```
const { tableProps, searchFormProps } = useTable<
    ITransactionDetails,
    HttpError,
    ITransactionFilterVariables
  >({
    metaData: {
      operation: 'get_history_transaction_finance',
    },
    queryOptions: {
      enabled: false,
    },
...
```

But the think is that, when the button of filter is fired, the table is not fetching data to my server.
Please help show me what I'm missing here.
Very appreciate of your help

Thanks

**Comment by vinclou** [comment]
@heriparid Could you share the full component, it's hard to tell only looking at the hook usage?

**Reply by heriparid** [reply]
Hi @vinclou ,
Well I came up with using state variable to assign "enableQuery" = false at initial load.
```
const [enableQuery, setEnableQuery] = useState(false);

  const { tableProps, searchFormProps } = useTable<
    ITransactionDetails,
    HttpError,
    ITransactionFilterVariables
  >({
    pagination: {
      pageSize: 5,
    },
    queryOptions: {
      enabled: enableQuery,
    },
```

But I encountered that on the first load the loading view is visible, and then my workaround to set as follow:
```
if (!enableQuery) {
    tableProps.loading = false;
  }
```

I'm not sure that the workaround is legitimate approach or not.

Thanks
